lib/qr_code_scanner.dart

```diff
@@ -0,0 +1,271 @@
+    super.didChangeDependencies();
+
+    widget.controller.qrCodeScannerAction = this;
+  }
+
+  @override
+  void didUpdateWidget(covariant QRCodeScanner oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    widget.controller.qrCodeScannerAction = this;
+  }
+
+  @override
+  Widget buildWidget(BuildContext context) {
+    return EnsembleBoxWrapper(
+      boxController: widget.controller,
+      widget: QRView(
+        key: qrKey,
+        cameraFacing: widget.controller.mode,
+        formatsAllowed: widget.controller.allFormatsAllowed,
+        overlayMargin: widget.controller.overlayMargin ?? EdgeInsets.zero,
+        overlay: QrScannerOverlayShape(
+          borderColor: widget.controller.cutOutBorderColor ?? Colors.red,
+          // borderColor: Colors.yellow,
+          overlayColor: widget.controller.overlayColor ??
+              const Color.fromRGBO(0, 0, 0, 80),
+          borderRadius: Utils.getDouble(widget.controller.cutOutBorderRadius,
+              fallback: 0),
+          borderLength: Utils.getDouble(widget.controller.cutOutBorderLength,
+              fallback: 40),
+          borderWidth: Utils.getDouble(widget.controller.cutOutBorderWidth,
+              fallback: 3.0),
+          cutOutHeight:
+              Utils.getDouble(widget.controller.cutOutHeight, fallback: 200),
+          cutOutWidth:
+              Utils.getDouble(widget.controller.cutOutWidth, fallback: 200),
+        ),
+        onQRViewCreated: _onQRViewCreated,
+        onPermissionSet: (ctrl, p) => _onPermissionSet(context, ctrl, p),
+      ),
+    );
+  }
+
+  void _onQRViewCreated(QRViewController controller) {
+    setState(() {
+      qrController = controller;
+    });
+
+    if (widget.controller.onInitialized != null) {
+      ScreenController().executeAction(
+        context,
+        widget.controller.onInitialized!,
+        event: EnsembleEvent(widget.controller),
+      );
+    }
+
+    qrController?.scannedDataStream.listen((scanData) {
+      // print('Code Data: ${scanData.code}');
+      // print('Format Data: ${scanData.format}');
+      // print('ToString Data: ${scanData.toString()}');
+      if (widget.controller.onReceived != null) {
+        final data = {
+          'format': scanData.format.name,
+          'data': scanData.code,
+          'rawBytes': scanData.rawBytes,
+        };
+
+        ScreenController().executeAction(
+          context,
+          widget.controller.onReceived!,
+          event: EnsembleEvent(widget.controller, data: data),
+        );
+      }
+    });
+  }
+
+  void _onPermissionSet(BuildContext context, QRViewController ctrl, bool p) {
+    log('${DateTime.now().toIso8601String()}_onPermissionSet $p');
+    if (!p) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(content: Text('no Permission')),
+      );
+    }
+  }
+
+  @override
+  void flipCamera() {
+    qrController?.flipCamera();
+  }
+
+  // TODO:
+  void getSystemFeatures() async {
```

Comment:
@vinothvino42 what is this for anyway?

Reply:
please also upload a video of this working

Reply:
This is to get the device capabilities for scanning a qrcode, whether its having back and front camera, flash, etc
https://pub.dev/documentation/qr_code_scanner/latest/qr_code_scanner/SystemFeatures-class.html

Reply:
> please also upload a video of this working

Sure, I'll upload and send you the link now